sar_pipeline/utils/sentinel1.py

```diff
@@ -2,21 +2,47 @@
 
 # Taken from https://sentiwiki.copernicus.eu/web/s1-products#S1Products-SARNamingConventionS1-Products-SAR-Naming-Convention
 SENTINEL_1_ID_PATTERN = (
-    r"^S1[ABC]_"  # MMM: S1A, S1B, S1C
+    r"(S1[ABC])_"  # MMM: S1A, S1B, S1C
     r"(S1|S2|S3|S4|S5|S6|IW|EW|WV|EN|N[1-6]|Z[IEW1-6])_"  # BB: Beam identifiers
     r"(RAW|SLC|GRD|OCN|ETA)"  # TTT: Product Type
-    r"[FHM_]_"  # R: Resolution class
-    r"[012A]"  # L: Processing level
-    r"[SANCX]"  # F: Product class
+    r"([FHM_])_"  # R: Resolution class
+    r"([012A])"  # L: Processing level
+    r"([SANCX])"  # F: Product class
     r"(SH|SV|DH|DV|HH|HV|VV|VH)_"  # PP: Polarisation
-    r"\d{8}T\d{6}_"  # Start datetime
-    r"\d{8}T\d{6}_"  # Stop datetime
-    r"\d{6}_"  # Orbit number (000000–999999)
-    r"(?!000000)[A-F0-9]{6}_"  # Data-take ID: non-zero hex
-    r"[A-F0-9]{4}"  # CRC16 and extension
+    r"(\d{8}T\d{6})_"  # Start datetime
+    r"(\d{8}T\d{6})_"  # Stop datetime
+    r"(\d{6})_"  # Orbit number (000000–999999)
+    r"((?!000000)[A-F0-9]{6})_"  # Data-take ID: non-zero hex
+    r"([A-F0-9]{4})"  # CRC16 and extension
```

Comment:
It might be good to do `r"([A-F0-9]{4})$"  # CRC16 and extension` with the dollar sign or a fullmatch below. Otherwise "S1A_IW_SLC__1SDV_20201019T191827_20201019T191854_034869_0410AD_29DE.stuff" is treated as valid, for example.